source/common/router/upstream_codec_filter.cc

```diff
@@ -23,17 +23,18 @@
 #include "source/common/http/headers.h"
 #include "source/common/http/message_impl.h"
 #include "source/common/http/utility.h"
+#include "source/common/stats/lazy_init.h"
```

Comment:
Why is this include needed all over the place? Should it be included in some other header so it's included when needed without needing to sprinkle it all over?

Reply:
a module previously includes "envoy/upstream/upstream.h" knows everything about a ClustertrafficStats,  but since LazyInit is defined under source/... per Josh's request, it has to include this header in someway. 

I made an attempt to include this header inside source/common/stats/utility.h which is usually good for many modules, but that creates a cyclic problem when runtime features_flag is a dep of lazy_init.h.

Reply:
Can we break the cycle by putting something into it's own header?

Reply:
I can do that if you feel strongly. I do think that we include the header is not a bad thing, which tells us the presence of Stats::LazyInit in this file.